src/components/img.tsx

```diff
@@ -47,18 +47,19 @@ interface Props {
 }
 
 const styles = (
-    role: Role,
+    image: Image,
     format: Format,
     supportsDarkMode: boolean,
 ): SerializedStyles => {
-    switch (role) {
+    switch (image.role) {
         case Role.Thumbnail:
             return css`color: ${neutral[60]};`;
         default:
             return css`
                 background-color: ${backgroundColour(format)};
                 color: ${neutral[60]};
                 ${darkModeCss(supportsDarkMode)`background-color: ${neutral[20]};`}
+                height: ${100 * image.height / image.width}%;
```

Comment:
This won't work. [Padding][] percentages are always based on the **`width`** of the containing block, but `height` is not

[Padding]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding#Values

Reply:
Good point, that's what you get for developing without actually getting a browser involved 🙈.

I'm tempted to wait to implement this until at least Storybook is up-and-running, and perhaps one or two of the consumer components as well 🤔. It may be that we'll always be able to specify the height from context, as we do [here](https://github.com/guardian/apps-rendering/blob/d1e6268b9ddcc28e0458be6c248ca7fda404f474/src/components/bodyImage.tsx#L25) for body images.

Reply:
These body images are also using a mix of percentage (`%`) and viewport-width (`vw`) values, which is probably not ideal.

Reply:
> which is probably not ideal

Can you elaborate on why this might be an issue?

Reply:
The percentage value is based on the [containing block][], whereas the viewport is based on the device. This means that 100% will only ever be the same as 100% if the containing block is taking up the full width of the viewport.

[containing block]: https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_Block

Reply:
Interesting point. I think previously the width _was_ `100%` of the viewport because on the app images bleed to the edge of the screen. But since dotcom doesn't do that, and we're trying to avoid design discrepancies with dotcom, it was changed to have the same padding as the text.

This _has_ caused an aspect ratio issue as you mentioned, but I suspect it wasn't noticed because of the `object-fit: cover` which [crops anything](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) that doesn't fit into the content box.

In summary, you're right and we should change this.